src/datachain/lib/dc/storage.py

```diff
@@ -23,7 +22,7 @@
 
 
 def from_storage(
-    uri: Union[str, os.PathLike[str], Iterable[str], Iterable[os.PathLike[str]]],
+    uri: Union[str, os.PathLike[str], list[str], list[os.PathLike[str]]],
```

Comment:
This should allow passing a mixed-type collections/iterable of `str` and `os.PathLike[str]` at the same time.